src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -80,6 +80,70 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    // Method Description:
+    // - Actually initialize the IconSource. Supports a variety of icons:
+    //    * If the icon is a path to an image, we'll use that.
+    //    * If it isn't, then we'll try and use the text as a FontIcon. If the
+    //      character is in the range of symbols reserved for the Segoe MDL2
+    //      Asserts, well treat it as such. Otherwise, we'll default to a Sego
+    //      UI icon, so things like emoji will work.
+    // - MUST BE CALLED ON THE UI THREAD.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    static void _refreshIcon(const Command& cmd)
```

Comment:
Wow I can't believe we let me ship this in the first place. Really, Command should just have an internal-only `hstring _iconPath`, and `Command::IconSource()` should probably do this dance to create the icon source. That's not really on you though, you're working with what I gave you, so I'm gonna sign off w/o this. Presumably this is what #7784 is talking about

Reply:
That's exactly what #7784 is haha. I'll start working on that. Dustin gave me a rundown of how it can work with converters.